objects/src/accounts/account_id.rs

```diff
@@ -0,0 +1,200 @@
+    pub fn new(seed: Word) -> Result<Self, AccountError> {
+        // hash the seed and build the ID from the 1st, 2nd, and 4th elements of the result
+        let hash = Hasher::hash_elements(&seed);
+        let id = Self(hash[0], hash[1], hash[3]);
+
+        // verify that the ID satisfies all rules
+        id.validate()?;
+
+        Ok(id)
+    }
+
+    // PUBLIC ACCESSORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns true if an account with this ID can issue fungible assets.
+    pub fn is_fungible_faucet(&self) -> bool {
+        self.tag() == Self::FUNGIBLE_FAUCET_TAG
+    }
+
+    /// Returns true if an account with this ID can issue non-fungible assets.
+    pub fn is_non_fungible_faucet(&self) -> bool {
+        self.tag() == Self::NON_FUNGIBLE_FAUCET_TAG
+    }
+
+    /// Returns true if an account with this ID can issue assets.
+    pub fn is_faucet(&self) -> bool {
+        self.is_fungible_faucet() || self.is_non_fungible_faucet()
+    }
+
+    // SEED GENERATORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Finds and returns a seed suitable for creating regular account IDs using the provided seed
+    /// as a starting point.
+    pub fn get_account_seed(_init_seed: [u8; 32]) -> Word {
+        todo!()
+    }
+
+    /// Finds and returns a seed suitable for creating account IDs for fungible faucets using the
+    /// provided seed as a starting point.
+    pub fn get_fungible_faucet_seed(_init_seed: [u8; 32]) -> Word {
+        todo!()
+    }
+
+    /// Finds and returns a seed suitable for creating account IDs for non-fungible faucets using
+    /// the provided seed as a starting point.
+    pub fn get_non_fungible_faucet_seed(_init_seed: [u8; 32]) -> Word {
+        todo!()
+    }
+
+    // HELPER METHODS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns the first bit of this account ID.
+    fn first_bit(&self) -> u8 {
+        (self.0.as_int() >> 63) as u8
+    }
+
+    /// Returns the last 32 bits of this account ID.
+    fn tag(&self) -> u32 {
+        self.2.as_int() as u32
+    }
+
+    /// Returns an error if:
+    /// - This account ID is for a fungible asset but the first bit of the ID is not ONE.
+    /// - There are fewer than 24 trailing ZEROs in this account ID.
+    fn validate(&self) -> Result<(), AccountError> {
+        if self.is_fungible_faucet() {
+            // IDs for fungible faucets must start with ONE
+            if self.first_bit() != 1 {
+                return Err(AccountError::fungible_faucet_id_invalid_first_bit());
+            }
+        } else {
+            // all account IDs must end with at least 24 ZEROs
+            if self.tag().trailing_zeros() < Self::MIN_TRAILING_ZEROS {
+                return Err(AccountError::account_id_too_few_trailing_zeros());
+            }
+        }
```

Comment:
Should we simplify this expression to
```suggestion
        // all account IDs must end with at least 24 ZEROs
        } else if self.tag().trailing_zeros() < Self::MIN_TRAILING_ZEROS {
            return Err(AccountError::account_id_too_few_trailing_zeros());
        }
```

Reply:
Fixed!